via/exceptions.py

```diff
@@ -10,7 +10,7 @@ class UpstreamServiceError(Exception):
     status_int = 409
 
 
-class UnhandledException(Exception):
+class UnhandledUpstreamException(Exception):
```

Comment:
Is this class really only used for upstream-related errors, and nothing else?

Reply:
Yep 100%. It's only raised as a result of `RequestException`'s. The name was overly broad before.